File: Silk4J/src/sample/insurance/Keywords.java
/**
 * Copyright 2013 Micro Focus. All rights reserved. 
 * Portions Copyright 1992-2009 Borland Software Corporation (a Micro Focus company).
 */
package sample.insurance;

import org.junit.Assert;

import com.borland.silk.keyworddriven.annotations.Argument;
import com.borland.silk.keyworddriven.annotations.Keyword;
import com.borland.silktest.jtf.BrowserBaseState;
import com.borland.silktest.jtf.Desktop;
import com.borland.silktest.jtf.common.BrowserType;
import com.borland.silktest.jtf.common.CommonOptions;
import com.borland.silktest.jtf.common.TruelogScreenshotMode;
import com.borland.silktest.jtf.common.types.FindOptions;
import com.borland.silktest.jtf.common.types.MouseButton;
import com.borland.silktest.jtf.xbrowser.BrowserApplication;
import com.borland.silktest.jtf.xbrowser.BrowserWindow;
import com.borland.silktest.jtf.xbrowser.DomButton;
import com.borland.silktest.jtf.xbrowser.DomElement;
import com.borland.silktest.jtf.xbrowser.DomLink;
import com.borland.silktest.jtf.xbrowser.DomListBox;
import com.borland.silktest.jtf.xbrowser.DomRadioButton;
import com.borland.silktest.jtf.xbrowser.DomTextField;

public class Keywords {
	private Desktop desktop = new Desktop();

	@Keyword("login as John Smith")
	public void login() {
		login("john.smith@gmail.com", "john");
	}

	@Keyword("login")
	public String login(String username, String password) {
		BrowserApplication bw = desktop.find("//BrowserApplication");
		bw.setActive();
		BrowserWindow browser = bw
				.find("//BrowserWindow");

		DomButton logout = browser.<DomButton> find(
				"//INPUT[@id='logout-form:logout']",
				new FindOptions(false, 500));
		if (logout != null)
			logout.select();

		browser.<DomTextField> find("//INPUT[@id='login-form:email']").setText(
				username);
		browser.<DomTextField> find("//INPUT[@id='login-form:password']")
				.setText(password);
		desktop.logInfo("Login", TruelogScreenshotMode.ActiveWindow);
		browser.<DomButton> find("//INPUT[@id='login-form:login']").select();
		desktop.logInfo("Logged In", TruelogScreenshotMode.ActiveWindow);

		return browser.<DomElement> find(
				"//LABEL[@class='login' and @for='logout-form']").getText();
	}

	@Keyword("fill out user contact data")
	public void fillOutCommunicationInfo(String zipCode, String email,
			Automobile type) {
		desktop.logInfo("CommunicationInfo", TruelogScreenshotMode.ActiveWindow);

		BrowserWindow browser = desktop
				.find("//BrowserApplication//BrowserWindow");
		browser.<DomTextField> find("//INPUT[@id='autoquote:zipcode']")
				.setText(zipCode);
		browser.<DomTextField> find("//INPUT[@id='autoquote:e-mail']").setText(
				email);
		browser.<DomRadioButton> find(
				"//INPUT[@id='autoquote:vehicle:" + type.ordinal() + "']")
				.select();
		desktop.logInfo("filled out CommunicationInfo",
				TruelogScreenshotMode.ActiveWindow);
		browser.<DomButton> find("//INPUT[@id='autoquote:next']").domClick();
	}

	@Keyword("fill out user details")
	public void fillOutGeneralInfo(@Argument("age") int age, Gender gender,
			DrivingRecord record) {
		desktop.logInfo("GeneralInfo", TruelogScreenshotMode.ActiveWindow);

		BrowserWindow browser = desktop
				.find("//BrowserApplication//BrowserWindow");
		browser.<DomTextField> find("//INPUT[@id='autoquote:age']",
				new FindOptions(10000)).setText("" + age);
		browser.<DomRadioButton> find(
				"//INPUT[@id='autoquote:gender:" + gender.ordinal() + "']")
				.select();
		browser.<DomRadioButton> find(
				"//INPUT[@id='autoquote:type:" + record.ordinal() + "']")
				.select();
		desktop.logInfo("filled out GeneralInfo",
				TruelogScreenshotMode.ActiveWindow);
		browser.<DomButton> find("//INPUT[@id='autoquote:next']").domClick();
	}

	@Keyword("fill out information about the car")
	public void fillOutCarInfo(int year, String make, String model,
			FinacialInfo info) {
		desktop.logInfo("CarInfo", TruelogScreenshotMode.ActiveWindow);

		BrowserWindow browser = desktop
				.find("//BrowserApplication//BrowserWindow");
		browser.<DomTextField> find("//INPUT[@id='autoquote:year']").setText(
				"" + year);
		browser.<DomElement> find("//IMG[@class='x-form-trigger x-for*']")
				.domClick();
		browser.<DomElement> find("//DIV[@textContents='" + make + "']")
				.domClick();
		browser.<DomElement> find("//IMG[@class='x-form-trigger x-for*'][2]")
				.domClick();
		browser.<DomElement> find("//DIV[@textContents='" + model + "']")
				.domClick();
		browser.<DomRadioButton> find(
				"//INPUT[@id='autoquote:finInfo:" + info.ordinal() + "']")
				.select();
		desktop.logInfo("filled out CarInfo",
				TruelogScreenshotMode.ActiveWindow);
		browser.<DomButton> find("//INPUT[@id='autoquote:next']").domClick();
	}

	@Keyword("start Browser")
	public BrowserApplication startBrowser() {
		return startBrowser(new BrowserBaseState());
	}

	@Keyword("start Browser with parameters")
	public BrowserApplication startBrowser(String url, BrowserType type) {
		return startBrowser(new BrowserBaseState(type, url));
	}

	public BrowserApplication startBrowser(BrowserBaseState baseState) {
		baseState.setOption(CommonOptions.OPT_APPREADY_TIMEOUT, 60000);
		if ("true".equals(System.getProperty("#sctm_tunnel_to_internal_proxy"))
				&& baseState.getBrowserType().equals(BrowserType.GoogleChrome)) {
			baseState.setCommandLineArguments("--proxy-server=localhost:8080");
			baseState.setCommandLinePattern("*");
		}
		return desktop.executeBaseState(baseState);
	}

	@Keyword("select Agent Lookup")
	public void selectAgentLookup() {
		selectAService("Agent Lookup");
	}

	@Keyword("select a Service")
	public void selectAService(String service) {
		desktop.<DomListBox> find(
				"//BrowserApplication//BrowserWindow//SELECT[@id='quick-link:jump-menu']")
				.select(service);
	}

	@Keyword("logout")
	public void logout() {
		desktop.<DomLink> find(
				"//BrowserApplication//BrowserWindow//A[@textContents='Home']")
				.click(MouseButton.LEFT);
		desktop.<DomButton> find(
				"//BrowserApplication//BrowserWindow//INPUT[@id='logout-form:logout']")
				.click(MouseButton.LEFT);
	}

	@Keyword("search for all Agents")
	public void searchForAllAgents() {
		desktop.<DomListBox> find(
				"//BrowserApplication//BrowserWindow//SELECT[@id='quick-link:jump-menu']")
				.select("Agent Lookup");
		desktop.<DomButton> find(
				"//BrowserApplication//BrowserWindow//INPUT[@id='show-all:search-all']")
				.click(MouseButton.LEFT);
	}

	@Keyword("validate that Agent Walker exists")
	public void validate_that_Agent_Walker_exists() {
		validate_that_Agent_exists("Walker");
	}

	@Keyword("validate that a specific Agent exists")
	public void validate_that_Agent_exists(String agentName) {
		desktop.logInfo("Validate if Agent '" +agentName +"' exists",
				TruelogScreenshotMode.ActiveWindow);
		Assert.assertTrue(desktop.exists(
				"//BrowserApplication//BrowserWindow//DIV[@textContents='"
						+ agentName + "']", 1000));
	}
}

File: Silk4NET/Keywords.cs
/**
 * Copyright 2013 Micro Focus. All rights reserved. 
 * Portions Copyright 1992-2009 Borland Software Corporation (a Micro Focus company).
 */

using SilkTest.Ntf;
using Silk.KeywordDriven;
using SilkTest.Ntf.XBrowser;
using Microsoft.VisualStudio.TestTools.UnitTesting;

public class Keywords
{
    public enum VehicleType
    {
        CAR, TRUCK
    }
    public enum Gender
    {
        MALE, FEMALE
    }
    public enum DrivingRecord
    {
        EXCELLENT, GOOD, FAIR, POOR
    }
    public enum FinancialInfo
    {
        OWN, FINANCED, LEASE
    }

    private readonly Desktop _desktop = Agent.Desktop;

    [Keyword("Start application", IsBaseState = true)]
    public void Start_application()
    {
        // Go to web page 'demo.borland.com/InsuranceWebExtJS'
        BrowserBaseState baseState = new BrowserBaseState();
        baseState.Execute();
    }

    [Keyword]
    public void Login(string email, string password)
    {
        BrowserWindow browserWindow = _desktop.BrowserWindow("Web");
        browserWindow.DomTextField("Login.EMail").SetText(email);
        browserWindow.DomTextField("Login.Password").SetText(password);
        browserWindow.DomButton("Login.Login").Click();
    }

    [Keyword("Login as default user")]
    public void SimpleLogin()
    {
        Login("john.smith@gmail.com", "john");
    }

    [Keyword("Select a service")]
    public void Select_a_service(string service)
    {
        BrowserWindow browserWindow = _desktop.BrowserWindow("Web");
        browserWindow.DomListBox("QuickJump").Select(service);
    }

    [Keyword("Search for all agents")]
    public void Search_for_all_agents()
    {
        BrowserWindow browserWindow = _desktop.BrowserWindow("Web");
        browserWindow.DomButton("SearchAll").Click();
    }

    [Keyword("Validate that Agent Walker exists")]
    public void Validate_that_Agent_Walker_exists()
    {
        BrowserWindow browserWindow = _desktop.BrowserWindow("Web");
        TestObject walker = browserWindow.Find("//DIV[@textContents='Walker']", new FindOptions(false, 5000));
        Assert.IsNotNull(walker);
    }

    [Keyword]
    public void Logout()
    {
        BrowserWindow browserWindow = _desktop.BrowserWindow("Web");
        browserWindow.DomLink("Home").Select();
        browserWindow.DomButton("Logout").Click();
    }

    [Keyword("Fill out user contact data")]
    public void Fill_out_user_contact_data(string zipCode, string email, VehicleType type)
    {
        BrowserWindow browserWindow = _desktop.BrowserWindow("Web");
        browserWindow.DomTextField("AutoQuote.ZipCode").SetText(zipCode);
        browserWindow.DomTextField("AutoQuote.EMail").SetText(email);

        switch (type)
        {
            case VehicleType.CAR:
                browserWindow.DomRadioButton("AutoQuote.Vehicle.Car").Select();
                break;
            case VehicleType.TRUCK:
                browserWindow.DomRadioButton("AutoQuote.Vehicle.Truck").Select();
                break;
        }
        browserWindow.DomButton("AutoQuote.Next").Click();
    }

    [Keyword("Fill out user details")]
    public void Fill_out_user_details(int age, Gender gender, DrivingRecord drivingRecord)
    {
        BrowserWindow browserWindow = _desktop.BrowserWindow("Web");
        browserWindow.DomTextField("AutoQuote.Age").SetText(age.ToString());
        switch (gender)
        {
            case Gender.MALE:
                browserWindow.DomElement("AutoQuote.Gender.Male").Click();
                break;
            case Gender.FEMALE:
                browserWindow.DomElement("AutoQuote.Gender.Female").Click();
                break;
        }

        switch (drivingRecord)
        {
            case DrivingRecord.EXCELLENT:
                browserWindow.DomElement("//INPUT[@id='autoquote:type:0']").Click();
                break;
            case DrivingRecord.GOOD:
                browserWindow.DomElement("//INPUT[@id='autoquote:type:1']").Click();
                break;
            case DrivingRecord.FAIR:
                browserWindow.DomElement("//INPUT[@id='autoquote:type:2']").Click();
                break;
            case DrivingRecord.POOR:
                browserWindow.DomElement("//INPUT[@id='autoquote:type:3']").Click();
                break;
        }

        browserWindow.DomButton("AutoQuote.Next").Click();
    }

    [Keyword("Fill out car details")]
    public void Fill_out_car_details(int year, string make, string type, FinancialInfo financialInfo)
    {
        BrowserWindow browserWindow = _desktop.BrowserWindow("Web");
        browserWindow.DomTextField("//INPUT[@id='autoquote:year']").SetText(year.ToString());
        
      
		browserWindow.DomElement("//IMG[@class='x-form-trigger x-for*']").DomClick();
		(browserWindow.Find(string.Format("//div[@textContents='{0}']", make)) as DomElement).Click();

		browserWindow.DomElement("//IMG[@class='x-form-trigger x-for*'][2]").DomClick();
		(browserWindow.Find(string.Format("//div[@textContents='{0}']", type)) as DomElement).Click();
		
        switch (financialInfo)
        {
            case FinancialInfo.OWN:
                browserWindow.DomRadioButton("//INPUT[@id='autoquote:finInfo:0']").Select();
                break;
            case FinancialInfo.FINANCED:
                browserWindow.DomRadioButton("//INPUT[@id='autoquote:finInfo:1']").Select();
                break;
            case FinancialInfo.LEASE:
                browserWindow.DomRadioButton("//INPUT[@id='autoquote:finInfo:2']").Select();
                break;
        }

        browserWindow.DomButton("AutoQuote.Next").Click();
    }

}